lib/iex/lib/iex/introspection.ex

```diff
@@ -557,26 +547,59 @@ defmodule IEx.Introspection do
       :error ->
         :no_beam
 
-      _ when is_nil(docs) ->
-        :no_docs
-
       {:ok, callbacks} ->
         docs =
-          docs
+          callbacks
+          |> Enum.map(&translate_callback/1)
           |> Enum.filter(filter)
-          |> Enum.map(fn
-            {{:macrocallback, fun, arity}, _, _, doc, metadata} ->
-              macro = {:"MACRO-#{fun}", arity + 1}
-              {format_callback(:macrocallback, fun, macro, callbacks), doc, metadata}
-
-            {{kind, fun, arity}, _, _, doc, metadata} ->
-              {format_callback(kind, fun, {fun, arity}, callbacks), doc, metadata}
+          |> Enum.sort_by(&elem(&1, 0), &>=/2)
```

Comment:
Since you are sorting on the first element of the tuple and by `&>=/2`, this is equivalent to `|> Enum.sort()`.

Reply:
Usually `sort/1` with `reverse/1` is faster than sorting with a custom function. That's not true in `sort_by/2` though since it sorts using a custom comparator anyway.

Reply:
Fixed